packages/sources/coinpaprika-test/src/crypto-utils.ts

```diff
@@ -0,0 +1,142 @@
+import { AdapterConfig } from '@chainlink/external-adapter-framework/config'
+import { customSettings, getApiEndpoint, getApiHeaders } from './config'
+import { SingleNumberResultResponse } from '@chainlink/external-adapter-framework/util'
+
+export const inputParameters = {
+  base: {
+    aliases: ['from', 'coin'],
+    description: 'The symbol of symbols of the currency to query',
+    required: true,
+    type: 'string',
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+  coinid: {
+    description: 'The coin ID (optional to use in place of `base`)',
+    required: false,
+    type: 'string',
+  },
+} as const
+
+export interface CryptoRequestBody {
+  quotes: string
+}
+
+export interface CryptoResponseSchema {
+  id: string
+  name: string
+  symbol: string
+  rank: number
+  circulating_supply: number
+  total_supply: number
+  max_supply: number
+  beta_value: number
+  first_data_at: string
+  last_updated: string
+  quotes: {
+    [key: string]: {
+      price: number
+      volume_24h: number
+      volume_24h_change_24h: number
+      market_cap: number
+      market_cap_change_24h: number
+      percent_change_15m: number
+      percent_change_30m: number
+      percent_change_1h: number
+      percent_change_6h: number
+      percent_change_12h: number
+      percent_change_24h: number
+      percent_change_7d: number
+      percent_change_30d: number
+      percent_change_1y: number
+      ath_price: number
+      ath_date: string
+      percent_from_price_ath: number
+    }
+  }
+  cost?: number
+}
+
+export interface CryptoRequestParams {
+  coinid?: string
+  base: string
+  quote: string
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: CryptoRequestParams
+  }
+  Response: SingleNumberResultResponse
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: CryptoRequestBody
+    ResponseBody: CryptoResponseSchema[]
+  }
+}
+
+const chunkArray = (params: CryptoRequestParams[], size = 3): CryptoRequestParams[][] =>
+  params.length > size ? [params.slice(0, size), ...chunkArray(params.slice(size), size)] : [params]
+
+export const buildBatchedRequestBody = (
+  params: CryptoRequestParams[],
+  config: AdapterConfig<typeof customSettings>,
+) => {
```

Comment:
Enforce return type

Reply:
i think for this one we need to export `ProviderRequestConfig` type from the framework

Reply:
Hmm I see. For now, can we do `HttpRequestConfig<never>[]` like we did in the coingecko EA?

Reply:
I think`HttpRequestConfig` is not available for newer version of the framework that coinpaprika is using.

Reply:
Yeah you're right. Do you think it makes sense to export `ProviderRequestConfig` from the framework? If so, could you get a PR up for that?

I'd just prefer enforcing types here to avoid future issues

Reply:
I think having explicit type would be great, tho in this particular case it doesn't change anything. Since every return property of `buildBatchedRequestBody` or `constructEntry` is stated clearly with their own types, even without having return type for the whole function if i change return structure to something wrong i will get an error for `prepareRequests` or `parseResponse` at the end .

Reply:
Ok makes sense. Let's skip this one and others that would clearly fail